python/pyspark/sql/session.py

```diff
@@ -760,6 +764,7 @@ def stop(self):
         """Stop the underlying :class:`SparkContext`.
         """
         self._sc.stop()
+        self._jvm.SparkSession.clearDefaultSession()
```

Comment:
hmm.. If we didn't set it in L231, perhaps we shouldn't clear it?
Or if we are picking up the JVM one in L217, we shouldn't clear it either?

WDYT?

Reply:
IIUC, since we already stop the jvm `SparkContext` with above line, it is not necessary to keep it.

Reply:
BTW, let me make a PR to your branch @jerryshao to deal with the failure soon. I was looking into this out of my curiosity.

Reply:
I'm also working on the failure, already figured out why.